CMakeLists.txt

```diff
@@ -1108,7 +1104,8 @@ _add_if("FTP"           NOT CURL_DISABLE_FTP)
 _add_if("FTPS"          NOT CURL_DISABLE_FTP AND SSL_ENABLED)
 _add_if("FILE"          NOT CURL_DISABLE_FILE)
 _add_if("TELNET"        NOT CURL_DISABLE_TELNET)
-_add_if("LDAP"          NOT CURL_DISABLE_LDAP)
+_add_if("LDAP"          NOT CURL_DISABLE_LDAP AND
+                        (CMAKE_USE_OPENLDAP OR USE_WIN32_LDAP))
```

Comment:
Isn't the original correct, my understanding is LDAP is just on by default even if no USE_OPENLDAP or USE_WIN32_LDAP? See https://github.com/curl/curl/blob/curl-7_50_1/lib/ldap.c#L27-L36 Also, if it is correct I'd prefer we determine features using the regular defines like USE_OPENLDAP instead of CMAKE_USE_OPENLDAP